src/wui/inputqueuedisplay.cc

```diff
@@ -525,20 +684,102 @@ void InputQueueDisplay::update_siblings_fill(int32_t delta) {
 		   std::max(0, std::min<int32_t>(static_cast<int32_t>(display->cache_max_fill_) + delta,
 		                                 display->cache_size_));
 		if (new_fill != display->cache_max_fill_) {
-			igb_.game().send_player_set_input_max_fill(
-			   building_, display->index_, display->type_, new_fill, settings_ != nullptr);
+			if (ib_.get_game()) {
+				ib_.game().send_player_set_input_max_fill(
+				   building_, display->index_, display->type_, new_fill, settings_ != nullptr);
+			} else {
+				if (settings_) {
```

Comment:
You have this code 3 times. Pull out a local function that will return a boolean and break out of the outer loop when it's true. Then you can also get rid of the ugly goto labels.